### ISSUE TITLE
Quick reply direct messages

### ISSUE THREAD

**Issue by luisjavierbautista** [tweepy/tweepy#1280]
Hi there,

I am trying to send quick reply options in a direct message (using api.send_direct_message function) but I can not find in documentation how I can pass these reply options. I only found quick_reply_type, so I want to validate if I am missing something or there is no support yet to use this by tweetpy. Maybe docs details about this could help other in the future.

Regards,

LJ

**Comment by om8007** [comment]
@luisjavierbautista @Harmon758 
#1364 will fix this issue by providing a parameter **quick_reply_options** which takes array of options objects(max 20 option).
Then function call will be like:
**API.send_direct_message(recipient_id, text, quick_reply_type = 'options', quick_reply_options = [{...
}, {...}, {...}])**
### Example
```
text = 'What is your favorite color?'
reply_options = [
            {
              "label": "Red",
              "description": "Color of heightened emotion, strength, and power.",
              "metadata": "external_id_1"
            },
            {
              "label": "Blue",
              "description": "Convey a sense of trust, loyalty, cleanliness, and understanding.",
              "metadata": "external_id_2"
            },
            {
              "label": "Black",
              "description": "Represents evil, darkness, night, and despair.",
              "metadata": "external_id_3"
            },
            {
              "label": "Pink",
              "description": "Represents friendship, affection, harmony and inner peace.",
              "metadata": "external_id_4"
            },
            {
              "label": "Green",
              "description": "Color of life, renewal, nature, and energy.",
              "metadata": "external_id_5"
            },
            {
              "label": "White",
              "description": "Associated with light, goodness, innocence, purity, and virginity.",
              "metadata": "external_id_6"
            }
         ]
API.send_direct_message(recipient_id, text, quick_reply_type = 'options', quick_reply_options = reply_options)
```
**P.S.** description and metadata are optional, see official documentation at [Options Quick Reply - Twitter Developers](https://developer.twitter.com/en/docs/direct-messages/quick-replies/api-reference/options)

**Comment by mahveotm** [comment]
Thanks, @om8007 for working on this, I really appreciate it. Please do you know if this can be integrated with [welcome message](https://developer.twitter.com/en/docs/direct-messages/welcome-messages/api-reference/get-welcome-message) and if Tweepy supports such. I think  @machine-uprising was working on the feature in 2019 but not too sure if they eventually made a PR. Marv

**Comment by hkucukali** [comment]
Hi, I am facing same issue currently. Is there any temporary solution that you can recommend until the next tweepy update? I don't know tweepy in detail but for example is there any currently available function that allows us to use event parameter as it is adviced in [#1081](https://github.com/tweepy/tweepy/issues/1081#issuecomment-413770522)

**Comment by Harmon758** [comment]
@om8007 As per my review of that PR, location and text input quick replies should be supported as well.

@mahveotm See #1310.

@hkucukali As you can see, that issue was closed by #1228 with https://github.com/tweepy/tweepy/commit/c4839adc7b9b0507ed4b7c5404f76bda8e6f51dd, so you can use `API.send_direct_message` without quick replies in v3.8.0 and newer.

**Comment by Harmon758** [comment]
@om8007 
Sorry, due to [Location and Text Input Quick Replies still being documented](https://twittercommunity.com/t/retired-location-and-text-input-quick-replies-are-still-documented/148102), I didn't realize that they had been retired from the API.

However, since Options Quick Replies are the only type now, `quick_reply_options` can simply replace `quick_reply_type` as a parameter, with the type set to "options" if options are passed, and with #1364 being stale, I've gone ahead and done that with https://github.com/tweepy/tweepy/commit/d8b02de5e27c91b477cc699a064507201f53a139.
